### ISSUE TITLE
Cannot check events emitted inside a constructor

### ISSUE THREAD

**Issue by MohamedLEGH** [rkalis/truffle-assertions#4]
I'm making ERC20 tokens and I emit a transfer event during the deployment of the contract (transfer of all tokens from address(0) to the owner) and I want to test that the event works well but it's look like it's not the case. (works fine with other blockchain calls like transfer or burn).

```
let instance = await Metacoin.new(param1);

truffleAssert.prettyPrintEmittedEvents(instance);
```
return 
     TypeError: Cannot read property 'length' of undefined

Same for truffleAssert.eventEmitted

**Comment by rkalis** [comment]
Hi @MohamedLEGH 

Thanks for presenting your issue. The reason of this error is that the contract constructor doesn't return a transaction result like most function calls do. Instead it returns an instance of the Metacoin contract in your case.

Luckily, this contract instance does include the hash of the transaction in which it was created, which we can use with web3 to reconstruct this transaction result object.

I think that these use cases should still be testable using `truffle-assertions` so I went ahead and extended its functionality.

I added a new function `truffleAssert.createTransactionResult(contract, transactionHash)` in version `0.4.0` of the library, which produces a transaction result object which can be used with the other functions.

I added usage documentation to the README for this new function, but in your case it should be:
```
let instance = await Metacoin.new(param1);
let result = await truffleAssert.createTransactionResult(instance, instance.transactionHash);

truffleAssert.prettyPrintEmittedEvents(result);
```

Let me know if this works for you. Good luck with your ERC20 token!
